crates/forge/tests/cli/test_cmd.rs

```diff
@@ -1048,3 +1048,58 @@ Traces:
 "#
     ]]);
 });
+
+const DETERMINISTIC_RANDOMNESS_TEST_CONTRACT: &str = r#"pragma solidity 0.8.24;
+import {Test, console2} from "forge-std/Test.sol";
+contract DeterministicRandomnessTest is Test {
+
+    function testDeterministicRandomUint() public {
+        uint256 x = vm.randomUint();
+        uint256 y = vm.randomUint();
+        assertEq(x, y);
+    }
+
+    function testDeterministicRandomUintRange(uint256 min, uint256 max) public {
+        vm.assume(max >= min);
+        uint256 x = vm.randomUint(min, max);
+        uint256 y = vm.randomUint(min, max);
+        assertEq(x, y);
+    }
+
+    function testDeterministicRandomAddress() public {
+        address x = vm.randomAddress();
+        address y = vm.randomAddress();
+        assertEq(x, y);
```

Comment:
I don't think this is what we want. The RNG instance should present in the `state` which is created once per test. Something like:
```rust
Cheatcodes {
  (private) rng: Option<StdRng>,
  pub fn rng(&mut self) -> &mut impl Rng {
    self.rng.get_or_insert(|| match self.config.seed /* ... */)
  }
}
```

Reply:
Makes sense, then we can reuse this in all the cheat codes.